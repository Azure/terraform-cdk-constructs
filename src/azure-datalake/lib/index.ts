export * from "./filesystem";
export * from "./path";
export * from "./datalake";
