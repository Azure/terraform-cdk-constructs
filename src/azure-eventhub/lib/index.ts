export * from "./authorization";
export * from "./eventhubcluster";
export * from "./consumer";
export * from "./instance";
export * from "./kusto-connection";
export * from "./namespace";
