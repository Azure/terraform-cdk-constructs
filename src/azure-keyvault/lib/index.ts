export * from './certificate'
export * from './key'
export * from './policy'
export * from './secret'
export * from './vault'

