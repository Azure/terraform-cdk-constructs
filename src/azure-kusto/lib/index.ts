export * from "./cluster";
export * from "./compute-specification";
export * from "./database";
