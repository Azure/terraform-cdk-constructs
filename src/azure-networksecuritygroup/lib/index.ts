export * from './preconfigured-rules';
export * from './securitygroup';
