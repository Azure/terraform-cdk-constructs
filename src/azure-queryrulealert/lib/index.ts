export * from './query-rule-alert'