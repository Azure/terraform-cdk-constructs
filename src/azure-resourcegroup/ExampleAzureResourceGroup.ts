import { AzureResourceGroup } from '.';
import { App, TerraformStack} from "cdktf";
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";
import { Construct } from 'constructs';
import { generateRandomString } from '../util/randomString';

const rndName = generateRandomString(10);

const app = new App();

export class exampleAzureResourceGroup extends TerraformStack {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    new AzurermProvider(this, "azure", {
      features: {},
    });
  
    new AzureResourceGroup(this, 'testRG', {
      name: `rg-${rndName}`,
      location: 'eastus',
      tags: {
          name: 'test',
          Env: "NonProd",
      },
      ignoreChanges: ['tags["Environment"]'],
      
    });

  }
}


new exampleAzureResourceGroup(app, "testAzureResourceGroup");
app.synth();