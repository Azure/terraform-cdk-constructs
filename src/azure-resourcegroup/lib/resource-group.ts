import { ResourceGroup } from "@cdktf/provider-azurerm/lib/resource-group";
import * as cdktf from "cdktf";
import { Construct } from "constructs";
import { AzureResource } from "../../core-azure/lib";

// Construct
/**
 * Properties for the resource group
 */
export interface GroupProps {
  /**
   * The Azure Region to deploy.
   */
  readonly location?: string;
  /**
   * The name of the Azure Resource Group.
   */
  readonly name?: string;

  /**
   * The tags to assign to the Resource Group.
   */
  readonly tags?: { [key: string]: string };
  /**
   * The lifecycle rules to ignore changes.
   */
  readonly ignoreChanges?: string[];
}

export class Group extends AzureResource {
  public readonly resourceGroupName: string;
  readonly props: GroupProps;
  idOutput: cdktf.TerraformOutput;
  locationOutput: cdktf.TerraformOutput;
  nameOutput: cdktf.TerraformOutput;

  public readonly id: string;
  public readonly location: string;
  public readonly name: string;

  constructor(scope: Construct, id: string, props: GroupProps = {}) {
    super(scope, id);

    this.props = props;

    const defaults = {
      name: props.name || `rg-${this.node.path.split("/")[0]}`,
      location: props.location || "eastus",
    };

    const azurermResourceGroupRg = new ResourceGroup(this, "rg", {
      ...defaults,
      tags: props.tags,
    });

    azurermResourceGroupRg.addOverride("lifecycle", [
      {
        ignore_changes: props.ignoreChanges || [],
      },
    ]);

    this.id = azurermResourceGroupRg.id;
    this.name = azurermResourceGroupRg.name;
    this.location = azurermResourceGroupRg.location;
    this.resourceGroupName = azurermResourceGroupRg.name;

    // Terraform Outputs
    this.idOutput = new cdktf.TerraformOutput(this, "id", {
      value: azurermResourceGroupRg.id,
    });
    this.locationOutput = new cdktf.TerraformOutput(this, "location", {
      value: azurermResourceGroupRg.location,
    });
    this.nameOutput = new cdktf.TerraformOutput(this, "name", {
      value: azurermResourceGroupRg.name,
    });

    /*This allows the Terraform resource name to match the original name. You can remove the call if you don't need them to match.*/
    this.locationOutput.overrideLogicalId("location");
    this.nameOutput.overrideLogicalId("name");
    this.idOutput.overrideLogicalId("id");
  }
}
