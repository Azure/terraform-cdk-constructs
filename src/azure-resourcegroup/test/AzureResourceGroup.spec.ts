import { Testing, TerraformStack} from 'cdktf';
import { exampleAzureResourceGroup} from './ExampleAzureResourceGroup'
import 'cdktf/lib/testing/adapters/jest';
import { AzureResourceGroup } from '../';
import {AzurermProvider} from "@cdktf/provider-azurerm/lib/provider";

describe('Resource Group With Defaults', () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", {features: {}});
    new AzureResourceGroup(stack, 'testRG');

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders a Resource Group with defaults and checks snapshot", () => {
    expect(
      Testing.synth(stack)
    ).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    expect(fullSynthResult).toPlanSuccessfully(); // Use the saved result
  });
});


describe('Resource Group Example', () => {
  
  it("renders the Example Resource Group and checks snapshot", () => {
    expect(
      Testing.synth(new exampleAzureResourceGroup(Testing.app(), "testAzexampleAzureResourceGroup"))
      
    ).toMatchSnapshot();
  });

  it("check if the produced terraform configuration is valid", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(new exampleAzureResourceGroup(Testing.app(), "testAzexampleAzureResourceGroup"))).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
  
    
    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(new exampleAzureResourceGroup(Testing.app(), "testAzexampleAzureResourceGroup"))).toPlanSuccessfully();
  });
});
