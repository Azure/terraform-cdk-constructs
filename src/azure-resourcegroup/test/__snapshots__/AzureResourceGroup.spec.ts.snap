// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource Group Example renders the Example Resource Group and checks snapshot 1`] = `
"{
  "output": {
    "id": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.id}"
    },
    "location": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.location}"
    },
    "name": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.name}"
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_resource_group": {
      "testRG_rg_D9B234FB": {
        "lifecycle": [
          {
            "ignore_changes": [
              "tags[\\"Environment\\"]"
            ]
          }
        ],
        "location": "eastus",
        "name": "rg-\${var.name}",
        "tags": {
          "Env": "NonProd",
          "name": "test"
        }
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.75.0"
      }
    }
  },
  "variable": {
    "name": {
      "default": "test",
      "description": "System name used to randomize the resources",
      "type": "string"
    }
  }
}"
`;

exports[`Resource Group With Defaults renders a Resource Group with defaults and checks snapshot 1`] = `
"{
  "output": {
    "id": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.id}"
    },
    "location": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.location}"
    },
    "name": {
      "value": "\${azurerm_resource_group.testRG_rg_D9B234FB.name}"
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_resource_group": {
      "testRG_rg_D9B234FB": {
        "lifecycle": [
          {
            "ignore_changes": [
            ]
          }
        ],
        "location": "eastus",
        "name": "rg-test"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.75.0"
      }
    }
  }
}"
`;
